include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -840,6 +840,171 @@ class __static_monotonic_dispatcher<::std::integer_sequence<::std::uint16_t, _X,
     }
 };
 
+struct __scalar_load_op
+{
+    oneapi::dpl::__internal::__pstl_assign __assigner;
+    template <typename _IdxType1, typename _IdxType2, typename _SourceAcc, typename _DestAcc>
+    void
+    operator()(_IdxType1 __idx_source, _IdxType2 __idx_dest, _SourceAcc __source_acc, _DestAcc __dest_acc) const
+    {
+        __assigner(__source_acc[__idx_source], __dest_acc[__idx_dest]);
+    }
+};
+
+template <std::uint8_t __vec_size>
+struct __vector_load
```

Comment:
I see this as optional / future TODO...
It seems like `__vector_load`, `__vector_store`, and possibly `__vector_walk`  are could rely upon on structure, with different arguments to implement them.
Perhaps there is another layer, but this is a pretty redundant pattern we are seeing here.  
Apply a function and index to each in a vector length, index, and set of arguments. 

Swap id explicitly different because of its structure, but the rest seem like they could be unified.

Reply:
Added a TODO comment here. It should be an easy change I can do after the milestone.